portal-api/src/v1/controllers/countries.controller.js

```diff
@@ -1,44 +1,122 @@
+ * @param {string} code - The country code.
+ * @returns {Promise<Object>} - A promise that resolves to the country information.
+ * @throws {Error} - If the API response is void or an error occurs while retrieving the country information.
+ */
+const getCountry = async (code) => {
+  try {
+    const response = await externalApi.countries.getCountry(code);
+
+    if (!response) {
+      throw new Error('Void response');
+    }
+
+    return response;
+  } catch (error) {
+    console.error('Unable to get country %o', error);
+    throw new Error('Unable to get country', { cause: error });
+  }
 };
 
-exports.findOneCountry = findOneCountry;
+/**
+ * Function to find one country.
+ *
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @returns {Object} - A promise that resolves with the response data.
+ * @throws {Error} - If unable to find one country.
+ */
+const findOne = async (req, res) => {
+  try {
+    const response = await getCountry(req.params.code);
 
-exports.findAll = async (req, res) => {
-  const countries = await externalApi.countries.getCountries();
-  const sortedCountries = sortCountries(countries);
-  res.status(200).send({
-    count: sortedCountries.length,
-    countries: sortedCountries,
-  });
+    if (!response) {
+      throw new Error('Void response');
+    }
+
+    const { status, data } = response;
+
+    if (data) {
+      return res.status(status).send(data);
+    }
+
+    return res.status(status).send({});
+  } catch (error) {
+    console.error('Unable to find one country %o', error);
+    throw new Error('Unable to find one country', { cause: error });
+  }
 };
 
-exports.findOne = async (req, res) => {
-  const response = await findOneCountry(req.params.code);
+/**
+ * Find all countries.
+ *
+ * This function retrieves all countries from an external API and sorts them alphabetically.
+ * It returns an object with the count of countries and the sorted array of countries.
+ * If there are no countries, it returns an empty array.
+ *
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @returns {Object} - An object with the count of countries and the sorted array of countries.
+ * @throws {Error} - If there is an error retrieving or sorting the countries.
+ */
+const findAll = async (req, res) => {
+  try {
+    const countries = await externalApi.countries.getCountries();
+    const sortedCountries = sortCountries(countries);
 
-  const { status, data } = response;
+    if (sortCountries.length) {
+      return res.status(200).send({
+        count: sortedCountries.length,
+        countries: sortedCountries,
+      });
+    }
 
-  if (data) {
-    return res.status(status).send(data);
+    return res.status(500).send({
+      count: 0,
+      countries: [],
+    });
+  } catch (error) {
+    console.error('Unable to find all countries %o', error);
+    throw new Error('Unable to find all countries', { cause: error });
```

Comment:
Considering this is a controller do we want throw errors here or do we want to handle them, returning the response (ala `sign-in-link.controller.js`)